src/frontend/src/handler/alter_owner.rs

```diff
@@ -37,71 +40,132 @@ pub async fn handle_alter_owner(
     let schema_path = SchemaPath::new(schema_name.as_deref(), &search_path, user_name);
 
     let new_owner_name = Binder::resolve_user_name(vec![new_owner_name].into())?;
-    let owner_id = session
-        .env()
-        .user_info_reader()
-        .read_guard()
-        .get_user_by_name(&new_owner_name)
-        .map(|u| u.id)
-        .ok_or(CatalogError::NotFound("user", new_owner_name))?;
-
-    let object = {
+    let (object, owner_id) = {
         let catalog_reader = session.env().catalog_reader().read_guard();
-        match stmt_type {
-            StatementType::ALTER_TABLE | StatementType::ALTER_MATERIALIZED_VIEW => {
-                let (table, schema_name) =
-                    catalog_reader.get_table_by_name(db_name, schema_path, &real_obj_name)?;
-                session.check_privilege_for_drop_alter(schema_name, &**table)?;
-                if table.owner() == owner_id {
-                    return Ok(RwPgResponse::empty_result(stmt_type));
+        let user_reader = session.env().user_info_reader().read_guard();
+        let new_owner = user_reader
+            .get_user_by_name(&new_owner_name)
+            .ok_or(CatalogError::NotFound("user", new_owner_name))?;
+        let owner_id = new_owner.id;
+        (
+            match stmt_type {
+                StatementType::ALTER_TABLE | StatementType::ALTER_MATERIALIZED_VIEW => {
+                    let (table, schema_name) =
+                        catalog_reader.get_table_by_name(db_name, schema_path, &real_obj_name)?;
+                    session.check_privilege_for_drop_alter(schema_name, &**table)?;
+                    let schema_id = catalog_reader
+                        .get_schema_by_name(db_name, schema_name)?
+                        .id();
+                    if !session.is_super_user()
+                        && !new_owner.is_super
+                        && !new_owner.check_privilege(
+                            &grant_privilege::Object::SchemaId(schema_id),
+                            AclMode::Create,
+                        )
```

Comment:
Add if the session user is super and the target schema is `public` we should also allow it.

Reply:
> Can we abstract the logic of this check into a function of session, so that other branches can also reuse it?
We can do this, but I think this function is useful only in `alter_owner.rs`, maybe use a in-file function instead?